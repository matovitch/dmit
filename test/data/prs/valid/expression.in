func f()
{
    5;
    2 + 3;
    2 + i;
    (3.14e+1);
    x + y * z;
    y * z + x;
    -a;
    f();
    f(2, 3 + 2);
    f(g());
    i >= 3;
}