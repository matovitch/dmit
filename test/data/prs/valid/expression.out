{"tree" :{"nodes":[{"kind":"END_OF_TREE","size":0,"childCount":0,"start":0,"stop":0},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":88,"stop":87},{"kind":"FUN_ARGUMENTS","size":0,"childCount":0,"start":87,"stop":84},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":82,"stop":81},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":79,"stop":77},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":77,"stop":75},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":75,"stop":74},{"kind":"EXP_BINOP","size":3,"childCount":3,"start":79,"stop":74},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":72,"stop":70},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":70,"stop":68},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":68,"stop":67},{"kind":"EXP_BINOP","size":3,"childCount":3,"start":72,"stop":67},{"kind":"LIT_DECIMAL","size":0,"childCount":0,"start":64,"stop":63},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":60,"stop":58},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":58,"stop":56},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":56,"stop":54},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":54,"stop":52},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":52,"stop":51},{"kind":"EXP_BINOP","size":3,"childCount":3,"start":56,"stop":51},{"kind":"EXP_BINOP","size":6,"childCount":3,"start":60,"stop":51},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":49,"stop":47},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":47,"stop":45},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":45,"stop":43},{"kind":"EXP_BINOP","size":3,"childCount":3,"start":49,"stop":43},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":43,"stop":41},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":41,"stop":40},{"kind":"EXP_BINOP","size":6,"childCount":3,"start":49,"stop":40},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":37,"stop":36},{"kind":"EXP_OPPOSE","size":1,"childCount":1,"start":38,"stop":36},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":34,"stop":33},{"kind":"FUN_CALL","size":1,"childCount":1,"start":34,"stop":31},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":29,"stop":28},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":27,"stop":26},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":24,"stop":22},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":22,"stop":20},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":20,"stop":19},{"kind":"EXP_BINOP","size":3,"childCount":3,"start":24,"stop":19},{"kind":"FUN_CALL","size":6,"childCount":3,"start":29,"stop":18},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":16,"stop":15},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":14,"stop":13},{"kind":"FUN_CALL","size":1,"childCount":1,"start":14,"stop":11},{"kind":"FUN_CALL","size":3,"childCount":2,"start":16,"stop":10},{"kind":"LIT_IDENTIFIER","size":0,"childCount":0,"start":8,"stop":6},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":6,"stop":4},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":4,"stop":3},{"kind":"EXP_BINOP","size":3,"childCount":3,"start":8,"stop":3},{"kind":"SCOPE","size":43,"childCount":11,"start":84,"stop":0},{"kind":"DEF_FUNCTION","size":46,"childCount":3,"start":90,"stop":0},{"kind":"DEFINITION","size":47,"childCount":1,"start":90,"stop":0},{"kind":"MODULE","size":48,"childCount":1,"start":90,"stop":0}]},"errors":[{"offset":4294967295,"errors":[]}]}